solutions/gold/cf-510C.mdx

```diff
@@ -0,0 +1,110 @@
+---
+id: cf-510C
+source: CF
+title: Fox And Names
+author: Daniel Ge
+---
+
+## Solution
+The key idea here is to perform topological sort on the characters of the alphabet. 
```

Comment:
There should be something that comes before this

Reply:
Should I add hints? Or maybe a leadup to thinking about topological sort?

Reply:
Yeah

Reply:
I have written hints and formatted them the way SansPapyrus [showed me](https://github.com/cpinitiative/usaco-guide/pull/3745) with the explanation and implementation in a solution spoiler as well. If you'd like me to do it in a different manner that's fine as well